yelp/UserYelpWebScraping.py

```diff
@@ -0,0 +1,236 @@
+        driver.execute_script("window.scrollTo(0, document.body.scrollHeight);")
+        time.sleep(2);
+        
+        
+        #get days of week from scrape
+        date = driver.find_elements(By.CLASS_NAME, "css-chan6m")
+        dayOfWeek = []
+        for i in range(len(date)):
+            if date[i].text.find("/")!=-1:
+                my_date = date[i].text
+                datetime_object = datetime.strptime(my_date, '%m/%d/%Y')
+                dayOfWeek.append(calendar.day_name[datetime_object.weekday()])  
+    
+        
+        #get text reviews and peoples names 
+        lists = driver.find_elements(By.XPATH, "//section[@aria-label='Recommended Reviews']//p[@class='comment__09f24__gu0rG css-qgunke']")
+        people = driver.find_elements(By.XPATH, "//section[@aria-label='Recommended Reviews']//a[@role='link']")
+
+        '''
+        #just for testing
+        if(len(lists)<2):
+            temp = len(lists)
+        else:
+            temp = 2
+        '''
+        
+        temp = len(lists)
+    
+        #interate reviews
+        for i in range(temp):
+            
+            name = people[i+1].text
+            if(checkExistance(name) == False):
+                
+                #fill database based on occasion/allergy
+                review = lists[i].text.lower()
+                current_day = dayOfWeek[i]
+                business = YelpApiCalls.return_business(rest_id)
+                
+                #i currently add the categories listed, not our umbrella terms. will switch next sprint
+                types = business.get('categories')
+                pos = []
+                for j in range(len(types)):
+                    pos.append(types[j].get('alias'))
+                neg = None
+                restrictions = []
+                
+                if review.find('lunch') != -1:
+                    meal = 'lunch'
+                elif review.find('breakfast') != -1 or review.find('brunch') != -1 or review.find('morning') != -1:
+                    meal = 'breakfast'
+                else:
+                    meal = 'dinner'
+      
+                occasion = None
+                
+                for res in restrictionsList:
+                    if(review.find(res)!=-1 and res not in restrictions):
+                        restrictions.append(res)
+                
+                if q in restrictionsList:
+                    for occ in occasions:
+                        if review.find(' ' + occ.replace('+', ' ') + ' ') != -1:
+                            q = occ
+                                
+                if q == "date+night":
+                    occasion = 'date'
+                    num_people = 2
+                            
+                elif q=='solo':
+                    occasion = 'solo'
+                    num_people = '1'
+                            
+                elif q=='friend':
+                    occasion = 'friends'
+                    num_people = 2
```

Comment:
jc why is the default number of people for the 'friends' occasion 2?

Reply:
If the person in the review said friend (singular), we would still call the occasion "friends", but we know it was just the reviewer and the single friend. If the review said friends, I assume it was the reviewer, and 2 or more other friends and randomly generate a number (you can see in the next elif)